### PR TITLE
If there are ports in a subnet, the interface cannot be deleted.

### PR DESCRIPTION
**Pull request by hmunfru**
#### Reviewers
@jframos @jesuspg

#### Description
https://jira.tid.es/browse/CLAUDIA-4857

#### Test
tested locally

### CODE REVIEW

File: core/src/main/java/com/telefonica/euro_iaas/paasmanager/claudia/impl/OpenstackNetworkClientImpl.java [hunk 1]
```diff
@@ -568,7 +568,16 @@ public SubNetworkInstance loadSubNetwork(ClaudiaData claudiaData, String subNetI
      */
     public void deleteNetworkToPublicRouter(ClaudiaData claudiaData, NetworkInstance netInstance, String region)
             throws InfrastructureException {
-        log.info("Delete Interfact from net " + netInstance.getNetworkName() + " to public router in region " + region);
+        log.info("Delete Interfact from net " + netInstance.getNetworkName() +
+            " to public router in region " + region);
+
+        List<Port> ports = listPortsFromNetwork(claudiaData, region, netInstance.getInterfaceNetId());
+        if (ports.size() > 0) {
+            String msm = "Error to delete the interface from public router  " + netInstance.getNetworkName() +
```
Comment (jesuspg) [inline core/src/main/java/com/telefonica/euro_iaas/paasmanager/claudia/impl/OpenstackNetworkClientImpl.java]: variable name "msm" ?

File: core/src/test/java/com/telefonica/euro_iaas/paasmanager/claudia/impl/OpenStackNetworkImplTest.java [hunk 2]
```diff
@@ -296,10 +297,19 @@ public void shouldDestroyRouter() throws OpenStackException, InfrastructureExcep
     public void shouldDeleteNetworkToPublicRouter() throws OpenStackException, InfrastructureException {
 
         // when
-        String response = "ok";
+        String response = "{\"ports\": [ {\"status\": \"ACTIVE\",\"name\": \"\",  \"admin_state_up\": true, \"network_id\": "
+            + "\"6a609412-3f04-485c-b269-b1a9b9ecb6bf\", \"tenant_id\": \"08bed031f6c54c9d9b35b42aa06b51c0\",\"binding:vif_type\": \"ovs\","
+            + "\"device_owner\": \"network:dhcp\", \"binding:capabilities\": {\"port_filter\": true}, \"mac_address\": \"fa:16:3e:54:1a:40\","
+            + "\"fixed_ips\": [ {\"subnet_id\": \"601ec829-a166-4b39-9c74-166cc76f251b\", \"ip_address\": \"172.31.0.3\"}],"
+            + "\"id\": \"07fd27d2-9ce1-48f3-be83-c7d2b7041a1a\", \"security_groups\": [], \"device_id\": \"dhcpfa3e6aae-2140-5176-877a-2f67684a3165-6a609412-3f04-485c-b269-b1a9b9ecb6bf\""
```
Comment (jesuspg) [inline core/src/test/java/com/telefonica/euro_iaas/paasmanager/claudia/impl/OpenStackNetworkImplTest.java]: line > 120 chars

File: core/src/test/java/com/telefonica/euro_iaas/paasmanager/claudia/impl/OpenStackNetworkImplTest.java [hunk 3]
```diff
@@ -308,6 +318,29 @@ public void shouldDeleteNetworkToPublicRouter() throws OpenStackException, Infra
...
+
+        when(openStackUtil.listPorts(any(PaasManagerUser.class), anyString())).thenReturn(response);
+        NetworkInstance net = new NetworkInstance("router", "vdc", "region");
+        SubNetworkInstance subNet = new SubNetworkInstance("dd", "vdc", "region", "ID");
+        subNet.setIdSubNet("ID");
+        net.addSubNet(subNet);
+        when(openStackUtil.deleteInterfaceToPublicRouter(any(PaasManagerUser.class), any(NetworkInstance.class),
+            anyString())).thenReturn(response);
+        openStackNetworkImpl.deleteNetworkToPublicRouter(claudiaData, net, REGION);
+
```
Comment (jesuspg) [inline core/src/test/java/com/telefonica/euro_iaas/paasmanager/claudia/impl/OpenStackNetworkImplTest.java]: add verify or assert

**Comment by jframos** [comment]
LGTM

**Comment by jesuspg** [comment]
LGTM

**Pull request merged**